src/main/java/com/zendesk/maxwell/replication/MaxwellReplicator.java

```diff
@@ -214,7 +238,6 @@ private RowMapBuffer getTransactionRows() throws Exception {
 				continue;
 			}
 
-			setReplicatorPosition((AbstractBinlogEventV4) v4Event);
```

Comment:
But a replicator restart could happen within `getTransactionRows`, which would mean:

 - we're in a transaction, processing rows
 - we reset the replicator position to the end of the previous transaction
 - we continue to try and process the `BEGIN` or other non-transactional events inside a transaction
 - 😿